File: easy/0643_maximum_average_subarray_i/solution.cpp
#include <numeric>
#include <vector>

class Solution {
public:
  double findMaxAverage(std::vector<int> &nums, int k) {
    int currSum = std::reduce(nums.begin(), nums.begin() + k);
    int maxSum = currSum;

    for (size_t i = k; i < nums.size(); ++i) {
      currSum = currSum + nums[i] - nums[i - k];
      maxSum = std::max(maxSum, currSum);
    }

    return (double)maxSum / k;
  }
};

File: easy/0643_maximum_average_subarray_i/solution.py
from typing import List


class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        currSum = sum(nums[i] for i in range(k))
        maxAverage = currSum / k

        for i in range(1, len(nums) - k + 1):
            currSum = currSum - nums[i - 1] + nums[i + k - 1]
            maxAverage = max(maxAverage, currSum / k)

        return maxAverage
